x/concentrated-liquidity/position.go

```diff
@@ -2,36 +2,53 @@ package concentrated_liquidity
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/gogo/protobuf/proto"
 
 	"github.com/osmosis-labs/osmosis/v12/osmoutils"
 	types "github.com/osmosis-labs/osmosis/v12/x/concentrated-liquidity/types"
 )
 
-// nolint: unused
-func (k Keeper) updatePositionWithLiquidity(ctx sdk.Context,
+func (k Keeper) initOrUpdatePosition(ctx sdk.Context,
 	poolId uint64,
 	owner sdk.AccAddress,
 	lowerTick, upperTick int64,
 	liquidityDelta sdk.Int,
-) {
-	position := k.getPosition(ctx, poolId, owner, lowerTick, upperTick)
+) (err error) {
+	position, err := k.GetPosition(ctx, poolId, owner, lowerTick, upperTick)
+	if err != nil {
+		return err
+	}
 
 	liquidityBefore := position.Liquidity
-	liquidityAfter := liquidityBefore.Add(liquidityDelta)
+	var liquidityAfter sdk.Int
+	if liquidityDelta.IsNegative() {
+		liquidityAfter = liquidityBefore.Sub(liquidityDelta)
+	} else {
+		liquidityAfter = liquidityBefore.Add(liquidityDelta)
+	}
```

Comment:
to be clear, this means that no matter what, we are adding liquidity to liquidityAfter, even if we have a negative delta right?

Reply:
So if we have a negative delta, we don't want to subtract it from liquidityBefore, so we do another negative to add the absolute value of liquidityDelta

Reply:
Yeah this seems odd to me. If it's negative, we should be subtracting, right?